HeterogeneousTest/AlpakaDevice/plugins/alpaka/AlpakaTestDeviceAdditionAlgo.dev.cc

```diff
@@ -0,0 +1,32 @@
+#include <cstdint>
+
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "HeterogeneousTest/AlpakaDevice/interface/alpaka/DeviceAddition.h"
+
+#include "AlpakaTestDeviceAdditionAlgo.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::HeterogeneousCoreAlpakaTestDevicePlugins {
+
+  struct KernelAddVectorsF {
+    template <typename TAcc>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc,
```

Comment:
In cases like this, where we have to keep the code inside the `ALPAKA_ACCELERATOR_NAMESPACE` because `wrapper_add_vectors_f` is not templated on the accelerator type, would it make sense to use `Acc1D` directly ?
```suggestion
    ALPAKA_FN_ACC void operator()(Acc1D const& acc,
```

Reply:
Actually, this could be a way to avoid templating all device code on the accelerator - which in turn could let us move some code back from header files to .dev.cc files...

Reply:
I agree the explicit `Acc1D` would be an avenue worth of exploring. I'll comment more in #44625.

Reply:
OK :-)
I'm testing the possibility on top of this PR... the CUDA version built fine, but I'm having problems with the ROCm version :-(

Reply:
The problem is that we are building CUDA and ROCm code in `__host__ __device__` mode: alpaka functions marked as `ALPAKA_FN_ACC` are marked as `__host__ __device__` when targetting the CUDA or ROCm backends.

This doesn't seem to cause any problems to CUDA, but HIP/ROCm is less forgiving, and fails to _link_ host-side libraries that contain kernels that make calls to `blockIdx.x`, _etc_.

I'll make a PR for cmsdist to improve the situation, and then a separate PR to test the `Acc1D` approach with these packages.